compiler/hash-ast/src/ast.rs

```diff
@@ -642,12 +781,18 @@ pub struct StrLitPat(pub Str);
 pub struct CharLitPat(pub char);
 
 /// An integer literal pattern.
-#[derive(Debug, PartialEq, Eq, Clone)]
-pub struct IntLitPat(pub u64);
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct IntLitPat {
+    pub value: u128,
```

Comment:
Maybe a function to get the "actual" rendered value of the integer? Maybe as a signed bigint or something?

Reply:
how do you mean?

Reply:
Like, if the integer is negative in the Hash code for example, the `value` field will appear like garbage if you try to print it. We should have a function to get a representable integer which is true to what it originally was in the source. That would make it appear correctly in `fmt`.

Reply:
Ah yes, it does need to be transformed properly

Reply:
Maybe this is the time to move this into its own crate like `hash_abi`?